R/utils-get_code_dependency.R

```diff
@@ -0,0 +1,530 @@
+# get_code_dependency ----
+
+#' Return the lines of code (with side-effects) needed to reproduce the object
+#'
+#' @details This function assumes that object relationships are established using the `<-`, `=`, or `->` assignment
+#' operators. It does not support other object creation methods like `assign` or `<<-`, nor non-standard-evaluation
+#' methods. To specify relationships between side-effects and objects, you can use the comment tag
+#' `# @linksto object_name` at the end of a line where the side-effect occurs.
+#'
+#' @param code An `expression` with `srcref` attribute or a `character` with the code.
+#' @param names A `character(n)` with object names.
+#'
+#' @return `character` vector of elements of `code` calls that were required to build the side-effects and
+#' influencing objects having and impact on the `object`
+#'
+#' @keywords internal
+get_code_dependency <- function(code, names) {
+  assert_classes(code, names)
+
+  if (is_empty(code)) return(code)
+
+  code <- assert_code(code) # Turns code into expression with srcref attribute.
```

Comment:
`assert_code` is an imprecise name for this function. Besides, code can't be an expression, we know it is always a character, so you can easily replace it with `parse`  

```suggestion
  code <- parse(code, keep.source = TRUE) 
```